juju/testing/conn.go

```diff
@@ -309,13 +310,17 @@ func (s *JujuConnSuite) OpenAPIAsNewMachine(c *gc.C, jobs ...state.MachineJob) (
 	if len(jobs) == 0 {
 		jobs = []state.MachineJob{state.JobHostUnits}
 	}
+
+	displayName := "foo"
+	instId := instance.Id(displayName)
```

Comment:
We should have display name and inst id different to each other we we are going to set display name. Otherwise leave display name empty. Is there need to set display name here at all? Most providers won't set it. We can set it when needed in those tests that do care.